moshi/src/main/java/com/squareup/moshi/BufferedSourceJsonReader.java

```diff
@@ -237,6 +238,13 @@
 
   private int doPeek() throws IOException {
     int peekStack = stack[stackSize - 1];
+
+    if (peekStack == JsonScope.EMPTY_DOCUMENT
+            && source.request(BYTE_ORDER_MARK.size())
+            && buffer.indexOf(BYTE_ORDER_MARK) == 0) {
```

Comment:
`rangeEquals` just got merged into Okio so once it's released this can become:
```java
if (peekStack == JsonScope.EMPTY_DOCUMENT && source.rangeEquals(0, BYTE_ORDER_MARK)) {
  source.skip(BYTE_ORDER_MARK.size());
}
```
which reads a lot nicer.